pandas/core/groupby/generic.py

```diff
@@ -234,9 +234,16 @@ def aggregate(
         relabeling = func is None
         columns = None
         no_arg_message = "Must provide 'func' or named aggregation **kwargs."
+        tuple_given_message = "func is expected but recieved {} in **kwargs."
         if relabeling:
             columns = list(kwargs)
-            func = [kwargs[col] for col in columns]
+            func = []
+            for col in columns:
+                if isinstance(kwargs[col], (list, tuple)):
```

Comment:
Why just these two types? Can we instead do `if not callable(kwargs[col])`?

Reply:
Yes, it makes more sense. I'll change it. Thanks for the suggestion.

Reply:
@TomAugspurger `aggfunc` can be `str` sometimes right. 
```
s.groupby(...).agg(one='sum',two='max)
```
Will `callable` return `True` the above example?

Reply:
can we factor this function out and put it in pandas/core/aggregation.py (and just call it from here).

this could be all the way from L238-249 maybe something like (this would be the code here)
```columns, kwargs = validate_func_kwargs(columns, kwargs)```

may not need all of the args / returns though

Reply:
@jreback Yes, we can that would be better option. 
Something like this.

```python
validate_func_kwargs(kwargs) -> columns, func
```
And incorporated all the changes suggested by other reviewers too. I'll add whatsnew too in an hour.